packages/cosmic-swingset/lib/ag-solo/start.js

```diff
@@ -234,27 +239,22 @@ export default async function start(basedir, withSES, argv) {
           if (broadcastJSON) {
             throw new Error(`duplicate type=http in connections.json`);
           }
-          broadcastJSON = makeHTTPListener(basedir, c.port, c.host, command);
+          broadcastJSON = makeHTTPListener(
+            basedir,
+            c.port,
+            c.host,
+            deliverInboundCommand,
+          );
           break;
         default:
           throw new Error(`unknown connection type in ${c}`);
       }
     }),
   );
 
-  const vatsDir = path.join(basedir, 'vats');
-  const stateDBDir = path.join(basedir, 'swingset-kernel-state');
-  const d = await buildSwingset(
-    stateDBDir,
-    mailboxStateFile,
-    withSES,
-    vatsDir,
-    argv,
-    broadcast,
-  );
-  ({ deliverInboundToMbx, deliverInboundCommand } = d);
-
   console.log(`swingset running`);
+  swingSetRunning = true;
```

Comment:
I'm guessing this `swingSetRunning` flag is here to prevent that race condition we talked about, where inbound messages might get delivered while the kernel is still catching up. If so, why does it only disable saving-state and delivering the outbound messages? Rather than somehow queueing the inbound messages until they can be delivered later.

Also, if we call `deliverOutbound()` once we're up and running, should we also do the save-state that was deferred?

Reply:
So this is an interesting question.  I've been assuming that a replay won't introduce new messages or state changes, because if it does, we will overwrite our old state and fail consensus.  If you'd like to do something different, we can.